### PR TITLE
new medal - vehicular manslaughter

### PR DESCRIPTION
**Pull request by Fikou**
## About the PR 

adds "Vehicular Manslaughter" medal for causing someone's death by ramming them with a pod/miniputt/minisub/car

this needs someone to actually add the medal to byond hub so
image:
image
need some ideas for the medal description

## Why's this needed? 

some fun new medals that wont hurt anyone (just kidding they will hurt the one getting rammed)

## Changelog

```
(u)Fikou:
(*)New medal, Vehicular Manslaughter, obtainable by... you probably already know how.
```

### CODE REVIEW

**Comment by Tarmunora** [comment]
Is this even possible to acquire? Direct damage doesn't cause death - that happens on life ticks - and there's no delay or anything. Do correct me if I'm wrong but I don't think this works currently

**Comment by Tarmunora** [comment]
Alternatively, grant the medal for a max-speed ramming (I think one of the engine upgrades grants a higher top speed, making in non-trivial)

File: code/modules/transport/pods/vehicle.dm [hunk 1]
```diff
@@ -581,8 +581,8 @@
 				SPAWN_DBG(0)
 					M.throw_at(throw_at, movement_controller:velocity_magnitude, 2)
 				logTheThing("combat", src, target, "crashes into [target] [log_loc(src)].")
-				if(isalive(M))
-					vehicular_manslaughter = 0 //we now check if person is alive after hit, if they are dead they get the achievement
+				if(M.health < 0)
```
Comment (Tarmunora) [inline code/modules/transport/pods/vehicle.dm:584]: Wont work as written - health is updated by the health update queue (which runs every half-second, give or take) 😓

Comment (Fikou) [reply code/modules/transport/pods/vehicle.dm:584]: god damn it

Comment (Tarmunora) [reply code/modules/transport/pods/vehicle.dm:584]: Also I think you had the conditional backwards here anyhow :shelterfrog:

Comment (Fikou) [reply code/modules/transport/pods/vehicle.dm:584]: i'll just take your suggestion of max speed ramming into a human seems less of a pain

Comment (pali6) [reply code/modules/transport/pods/vehicle.dm:584]: I don't think that's the case?

Comment (pali6) [reply code/modules/transport/pods/vehicle.dm:584]: oh wait nvm, humans don't call their parents

**Comment by Fikou** [comment]
> Alternatively, grant the medal for a max-speed ramming (I think one of the engine upgrades grants a higher top speed, making it non-trivial to do)

i was looking through the code to see if i could make this the way to obtain the medal and i found that the speedmod var from engines is actually not used at ALL for vehicles, they only check if the engine exists so they can fly, only this weird cruiser thing thats not a vehicle checks for speedmod of the engine
so basically helios engine saying "a really fast engine" is just a flatout lie
this is tragic

**Comment by Tarmunora** [comment]
Scope creep time :)

File: code/modules/transport/pods/vehicle.dm [hunk 2]
```diff
@@ -578,6 +581,12 @@
 				SPAWN_DBG(0)
 					M.throw_at(throw_at, movement_controller:velocity_magnitude, 2)
 				logTheThing("combat", src, target, "crashes into [target] [log_loc(src)].")
+				SPAWN_DBG(2.5 SECONDS)
+					if(M.health < 0) 
+						vehicular_manslaughter = 0 //we now check if person was sent into crit after hit, if they did we get the achievement
```
Comment (pali6) [inline code/modules/transport/pods/vehicle.dm:586]: This does the opposite of what you want (gives the medal only when you *don't* put the person into crit).

**Comment by pali6** [comment]
Someone who can add medals please merge if you approve.

**Pull request merged**